prow/test/integration/test/setup.go

```diff
@@ -0,0 +1,89 @@
+// +build e2etest
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package integration
+
+import (
+	"crypto/rand"
+	"crypto/sha256"
+	"flag"
+	"fmt"
+	"os"
+	"os/user"
+	"path"
+	"strings"
+	"testing"
+
+	_ "k8s.io/client-go/plugin/pkg/client/auth/oidc"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	ctrlruntimeclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	defaultNamespace = "default"
+	testpodNamespace = "test-pods"
+)
+
+var clusterContext = flag.String("cluster", "kind-kind-prow-integration", "The context of cluster to use for test")
+
+func getClusterContext() string {
+	return *clusterContext
+}
+
+func getDefaultKubeconfig(cfg string) string {
```

Comment:
all of this is something the `clientcfg.ConfigLoader` already does with its default ruleset

Reply:
Good to learn, done